src/sortStudents.ts

```diff
@@ -1,16 +1,96 @@
-
 export interface Student {
-  // describe Student interface
+  name: string;
+  surname: string;
+  age: number;
+  married: boolean;
+  grades: number[]
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name = 'name',
+  Surname = 'surname',
+  Age = 'age',
+  Married = 'married',
+  AverageGrade = 'grades'
+}
+
+export type SortOrder = 'asc' | 'desc';
+
+export function sortGrades(
+  array: Student[],
+  order: SortOrder,
+  field: 'grades',
+): Student[] {
+  function getAverageGrades(grades: number[]): number {
+    return grades.reduce((a: number, b: number) => a + b, 0) / grades.length;
+  }
+
+  const copy = [...array];
+
+  if (order === 'asc') {
+    return copy
+      .sort((s1: Student, s2: Student) => getAverageGrades(s1[field])
+      - getAverageGrades(s2[field]));
+  }
+
+  return copy
+    .sort((s1: Student, s2: Student) => getAverageGrades(s2[field])
+    - getAverageGrades(s1[field]));
+}
+
+export function sortStrings(
+  array: Student[],
+  order: SortOrder,
+  field: 'name' | 'surname',
+): Student[] {
+  const copy = [...array];
+
+  if (order === 'asc') {
+    return copy
+      .sort((s1: Student, s2: Student) => s1[field].localeCompare(s2[field]));
+  }
+
+  return copy
+    .sort((s1: Student, s2: Student) => s2[field].localeCompare(s1[field]));
 }
 
-// create SortOrder type
-export type SortOrder;
+export function sortNumbers(
+  array: Student[],
+  order: SortOrder,
+  field: 'age' | 'married' | 'grades',
+): Student[] {
+  const copy = [...array];
+
+  if (order === 'asc') {
+    return copy
+      .sort((s1: Student, s2: Student) => +s1[field] - +s2[field]);
+  }
+
+  return copy.sort((s1: Student, s2: Student) => +s2[field] - +s1[field]);
+}
+
+export function sortStudents(
+  students: Student[],
+  sortBy: keyof SortType,
+  order: SortOrder,
+): Student[] {
+  switch (sortBy) {
+    case 'name':
```

Comment:
use enum values in cases
```suggestion
    case SortType.Name:
```